tests/misc_tests/test_collections.py

```diff
@@ -0,0 +1,212 @@
+from __future__ import print_function
+from __future__ import unicode_literals
+from __future__ import division
+from __future__ import absolute_import
+from builtins import *  # NOQA
+from future import standard_library
+standard_library.install_aliases()
+
+import collections
+import timeit
+import unittest
+
+from chainer import testing
+import numpy as np
+from scipy import special
+
+from chainerrl.misc.collections import _sample_n_k
+from chainerrl.misc.collections import RandomAccessQueue
+
+
+@testing.parameterize(
+    {'n': 2, 'k': 2},
+    {'n': 5, 'k': 1},
+    {'n': 5, 'k': 4},
+    {'n': 7, 'k': 2},
+    {'n': 20, 'k': 10},
+    {'n': 100, 'k': 5},
+)
+class TestSampleNK(unittest.TestCase):
+    def test_fast(self):
+        self.samples = [_sample_n_k(self.n, self.k) for _ in range(200)]
+        self.subtest_constraints()
+
+    def subtest_constraints(self):
+        for s in self.samples:
+            self.assertEqual(len(s), self.k)
+
+            all(0 <= x < self.n for x in s)
+
+            # distinct
+            t = np.unique(s)
+            self.assertEqual(len(t), self.k)
+
+    @testing.attr.slow
+    @testing.condition.repeat_with_success_at_least(3, 2)
+    def test_slow(self):
+        self.samples = [_sample_n_k(self.n, self.k) for _ in range(100000)]
+        self.subtest_total_counts()
+        self.subtest_order_counts()
+
+    def subtest_total_counts(self):
+        if self.k == self.n:
+            return
+
+        cnt = np.zeros(self.n)
+        for s in self.samples:
+            for x in s:
+                cnt[x] += 1
+
+        m = len(self.samples)
+
+        p = self.k / self.n
+        mean = m * p
+        std = np.sqrt(m * p * (1 - p))
+
+        self.subtest_normal_distrib(cnt, mean, std)
+
+    def subtest_order_counts(self):
+        if self.k < 2:
+            return
+
+        ordered_pairs = [(i, j) for j in range(self.k) for i in range(j)]
+        cnt = np.zeros(len(ordered_pairs))
+
+        for s in self.samples:
+            for t, (i, j) in enumerate(ordered_pairs):
+                if s[i] < s[j]:
+                    cnt[t] += 1
+
+        m = len(self.samples)
+
+        mean = m / 2
+        std = np.sqrt(m / 4)
+
+        self.subtest_normal_distrib(cnt, mean, std)
+
+    def subtest_normal_distrib(self, xs, mean, std):
+        n = len(xs)
+
+        sorted_xs = np.sort(xs)
+        cdfs = (1 + special.erf((sorted_xs - mean)
+                                / (np.sqrt(2) * std))) / 2
+        cdfs_l = (np.arange(n)) / n
+        cdfs_r = (np.arange(n) + 1) / n
+
+        # Kolmogorov-Smirnov statistic
+        d = max(np.amax(np.abs(cdfs - cdfs_x)) for cdfs_x in [cdfs_l, cdfs_r])
```

Comment:
scipy has `scipy.stats.kstest`, so I think it's better to use it.

You can get p-value by `scipy.stats.kstest(xs, 'norm', args=(mean, std))[1]`.